validator/modules/onnx/__init__.py

```
@@ -0,0 +1,289 @@
+                revision=revision,
+                cache_dir=self.config.output_dir,
+            )
+
+            self.model = ort.InferenceSession(model_path)
+
+            print(f"Successfully loaded ONNX model from {model_repo_id}")
+            return model_path
+
+        except Exception as e:
+            print(f"Error loading model from {model_repo_id}: {e}")
+            raise ValueError(f"Failed to load ONNX model: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)